.circleci/config.yml

```diff
@@ -0,0 +1,129 @@
+
+commands:
+  init:
+    description: "Install"
+    steps:
+      - kubernetes/install
+      - aws-eks/update-kubeconfig-with-authenticator:
+          authenticator-release-tag: v0.5.1
+          cluster-name: ${AWS_EKS_CLUSTER_NAME}
+          cluster-authentication-role-arn: ${AWS_EKS_ROLE_ARN}
+      - helm/install-helm-client:
+          version: v3.3.4
+      - run:
+          name: Install helmfile
+          command: |
+            curl -L --fail --retry 3 -o ./helmfile https://github.com/roboll/helmfile/releases/download/v0.130.0/helmfile_linux_amd64 
+            chmod +x ./helmfile
+            sudo mv ./helmfile /usr/local/bin/helmfile 
+      - run:
+          name: Install helm plugins
+          command: |
+            helm plugin install https://github.com/databus23/helm-diff
+      - checkout:
+          path: ./helmfile
+
+jobs:
+  lint:
+    executor: aws-eks/python3
+    steps:
+      - init
+      - run:
+          name: helmfile lint
+          working_directory: ./helmfile
+          command: |
+            helmfile -e ${TARGET_NAMESPACE} lint
+  validate:
+    executor: aws-eks/python3
+    steps:
+      - init
+      - run:
+          name: helmfile dry-run
+          working_directory: ./helmfile
+          command: |
+            helmfile -e ${TARGET_NAMESPACE} apply --args --dry-run --suppress-secrets --detailed-exitcode=false
+  deploy:
+    executor: aws-eks/python3
+    steps:
+      - init
+      - run:
+          name: helmfile apply
+          working_directory: ./helmfile
+          command: |
+            helmfile -e ${TARGET_NAMESPACE} apply --suppress-secrets --detailed-exitcode=false
+  test:
+    executor: aws-eks/python3
+    steps:
+      - init
+      - setup_remote_docker:
+          version: 19.03.13
+          docker_layer_caching: true
+      - run:
+          name: helm test
+          command: |
+            helm test tx-crafter -n ${TARGET_NAMESPACE}
+      - run:
+          name: Extract report from volume
```

Comment:
This step looks pretty complex. Isn't it possible to mount the volume to store the report before starting the test and then extract the artefact, instead of launching a new container just to do that?

Reply:
The volume is mounted before the test. Unfortunately there is no way to retrieve a file from a volume if its not bind to a pod :/. And I need the test pod to end to know the result of it. So didnt find better solution